src/app/components/Pagination.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+  <div class="flex items-center items-center justify-center py-6">
+    <div
+      v-if="currentPage !== 1"
+      @click="changePage(currentPage - 1)"
+      class="arrow arrow-left py-1 px-2 mx-1 bg-indigo-300 hover:bg-indigo-600 with-transition rounded-full cursor-pointer"
+    ></div>
+
+    <div v-for="(page, idx) in pagesArray" :key="idx">
+      <p
+        v-if="typeof page === 'number'"
+        @click="changePage(page)"
+        :class="paginationClassObject(page)"
+      >{{ page }}</p>
+      <p v-else-if="page === 'dots' && shoultRenderDots(idx)" class="py-1 px-2 mx-1">...</p>
+    </div>
+
+    <div
+      v-if="currentPage !== pagesCount"
+      @click="changePage(currentPage + 1)"
+      class="arrow arrow-right py-1 px-2 mx-1 bg-indigo-300 hover:bg-indigo-600 with-transition rounded-full cursor-pointer"
+    ></div>
+  </div>
+</template>
+
+<script>
+import { mapActions } from 'vuex';
+
+export default {
+  props: {
+    limit: Number,
+    offset: Number,
+    total: Number
+  },
+
+  computed: {
+    pagesCount() {
+      return Math.floor(this.total / this.limit) + 1;
+    },
+
+    currentPage() {
+      return Number(this.$route.query.page) || 1;
+    },
+
+    pagesArray() {
+      if (this.pagesCount <= 10) {
```

Comment:
Thinking about this method I would say that it could be like a reduce method for the initial array
The second action with it is to shrink long sequences with three dots
1. find amount of pages, let's say 10
2. create an initial array [2, 3, 4, 5, 6, 7, 8, 9]. We will keep first and last as a fast navigation
3. reduce array with "distance" count from the current page. For example if it's page 7 so the result should be like [..., 5, 6, **_7_**, 8, 9] or if it's 5 then [..., 3, 4, **_5_**, 6, 7, ...]
4. add < > regarding the current page, first and last pages to the array
Why?
- the declarative way is more flexible and understandable. If I as a user will ask you to change a distance for the current page you will have to change the whole method
- hardcode is always bad :)
- should be less code :)